File: app/src/main/java/com/donald/musictheoryapp/Utils/Button/ImageButton.java
package com.donald.musictheoryapp.Utils.Button;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.LayerDrawable;
import android.util.TypedValue;
import android.view.Gravity;

import androidx.core.content.ContextCompat;
import androidx.core.graphics.drawable.RoundedBitmapDrawable;
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory;

import com.donald.musictheoryapp.R;

@SuppressLint("ViewConstructor")
public class ImageButton extends QuestionButton
{
	public ImageButton(Context context, RoundedBitmapDrawable background)
	{
		super(context, background);
		TypedValue radius = new TypedValue();
		getResources().getValue(R.dimen.question_button_corner_radius, radius, true);
		background.setCornerRadius(radius.getDimension(getResources().getDisplayMetrics()));
		setElevation(context.getResources().getDimension(R.dimen.image_button_elevation));
	}
}

File: app/src/main/java/com/donald/musictheoryapp/customview/questionbutton/ImageButton.kt
package com.donald.musictheoryapp.customview.questionbutton

import android.annotation.SuppressLint
import android.content.Context
import androidx.core.graphics.drawable.RoundedBitmapDrawable
import android.widget.TextView
import android.view.LayoutInflater
import com.donald.musictheoryapp.R
import android.util.TypedValue

@SuppressLint("ViewConstructor")
class ImageButton(context: Context, backgroundBase: RoundedBitmapDrawable) : QuestionButton(
    context,
    backgroundBase
) {

    fun setNumber(number: Int) {
        val numberView = LayoutInflater.from(context).inflate(
            R.layout.part_button_number, this, false
        ) as TextView
        numberView.text = number.toString()
        val numberParams = LayoutParams(
            LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT
        )
        numberParams.addRule(ALIGN_PARENT_TOP)
        numberParams.addRule(ALIGN_PARENT_START)
        numberView.layoutParams = numberParams
        addView(numberView)
    }

    init {
        val radius = TypedValue()
        resources.getValue(R.dimen.question_button_corner_radius, radius, true)
        backgroundBase.cornerRadius = radius.getDimension(resources.displayMetrics)
        elevation = context.resources.getDimension(R.dimen.image_button_elevation)
    }
}